app/src/pages/inside/dashboardItemPage/newWidgets/tooltip/tooltip.jsx

```diff
@@ -0,0 +1,65 @@
+import { Component } from 'react';
+import { render } from 'react-dom';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import styles from './tooltip.scss';
+
+const cx = classNames.bind(styles);
+
+export class Tooltip extends Component {
+  static propTypes = {
+    launchName: PropTypes.string,
+    launchNumber: PropTypes.string,
+    startTime: PropTypes.string,
+    itemCases: PropTypes.number,
+    itemName: PropTypes.string,
+    color: PropTypes.string.isRequired,
+  };
+
+  static defaultProps = {
+    launchName: '',
+    launchNumber: '',
+    startTime: '',
+    itemCases: 0,
+    itemName: '',
+  };
+
+  dateFormat = (val) => {
```

Comment:
I guess this function better to place outside of component, because it's a pure function and not related with `Tooltip` directly.